server.py

```diff
@@ -0,0 +1,108 @@
+from sys import argv
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from http.cookies import SimpleCookie
+from cgi import FieldStorage
+
+from jinja2 import Environment, FileSystemLoader
+
+
+HOST = '127.0.0.1'
+PORT = 8001
+
+if len(argv) > 1:
+    PORT = int(argv[1])
+
+
+class Handler(BaseHTTPRequestHandler):
+
+    def _get_cookies(self):
+        cookies = SimpleCookie(self.headers.get('Cookie'))
+
+        if len(cookies) > 0 and int(cookies['auth'].value) > 0:
+            return True
+
+        return False
+
+    def _get_template(self, page):
+        file_loader = FileSystemLoader('templates')
+        env = Environment(loader=file_loader)
+        return env.get_template(page)
+
+    def _set_cookies(self, value):
+        cookies = SimpleCookie()
+        cookies['auth'] = value
+        self.send_header('Set-Cookie', cookies.output(header='', sep=''))
+
+    def _charge(self):
+        pass
+
+    def _error(self):
+        pass
+
+    def _form(self):
+        pass
+
+    def _log_in(self):
+        self._set_cookies(100)
+
+    def _log_out(self):
+        self._set_cookies(0)
```

Comment:
it's better to remove session cookie than set it to 0 (more typical)

Reply:
fixed